File: app/src/main/java/com/geovanni/banks/views/adapters/BanksAdapter.java
package com.geovanni.banks.views.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.geovanni.banks.R;
import com.geovanni.banks.bussiness.models.ServiceBanksResponse;
import com.squareup.picasso.Picasso;

import java.util.List;

public class BanksAdapter extends RecyclerView.Adapter<BanksAdapter.ViewHolder> {

    private List<ServiceBanksResponse> listBanks;
    private Context context;

    public BanksAdapter(Context context) {
        this.context = context;
    }

    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_bank, parent, false);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ServiceBanksResponse bank = listBanks.get(position);
        holder.txvName.setText(bank.getBankName());
        holder.txvDescription.setText(bank.getDescription());
        holder.txvAge.setText(bank.getAge());

        Picasso.get()
                .load(bank.getUrl())
                .error(R.drawable.ic_banks_splash)
                .placeholder(R.drawable.ic_banks_splash)
                .fit().centerCrop()
                .into(holder.imvBank);

    }

    @Override
    public int getItemCount() {
        return listBanks.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        private ImageView imvBank;
        private TextView txvName, txvDescription, txvAge;

        public ViewHolder(View itemView) {
            super(itemView);

            imvBank = (ImageView) itemView.findViewById(R.id.imvBank);
            txvName = (TextView) itemView.findViewById(R.id.txvName);
            txvDescription = (TextView) itemView.findViewById(R.id.txvDescription);
            txvAge = (TextView) itemView.findViewById(R.id.txvAge);

        }
    }

    public void replaceData(List<ServiceBanksResponse> listBanks) {
        if (listBanks != null) {
            this.listBanks = listBanks;
        }
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/geovanni/banks/views/adapters/BanksAdapter.kt
package com.geovanni.banks.views.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.geovanni.banks.R
import com.geovanni.banks.bussiness.models.ServiceBanksResponse
import com.squareup.picasso.Picasso

class BanksAdapter(private val context: Context) : RecyclerView.Adapter<BanksAdapter.ViewHolder>() {

    private var listBanks: List<ServiceBanksResponse>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_bank, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val bank = listBanks!![position]
        holder.txvName.text = bank.bankName
        holder.txvDescription.text = bank.description
        holder.txvAge.text = bank.age
        Picasso.get()
            .load(bank.url)
            .error(R.drawable.ic_banks_splash)
            .placeholder(R.drawable.ic_banks_splash)
            .fit().centerCrop()
            .into(holder.imvBank)
    }

    override fun getItemCount(): Int {
        return listBanks!!.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val imvBank: ImageView = itemView.findViewById<View>(R.id.imvBank) as ImageView
        val txvName: TextView = itemView.findViewById<View>(R.id.txvName) as TextView
        val txvDescription: TextView = itemView.findViewById<View>(R.id.txvDescription) as TextView
        val txvAge: TextView = itemView.findViewById<View>(R.id.txvAge) as TextView
    }

    fun replaceData(listBanks: List<ServiceBanksResponse>?) {
        if (listBanks != null) {
            this.listBanks = listBanks
        }
        notifyDataSetChanged()
    }

}